TASK: Discourage `DisableNodeAggregate` and remove duplicated command handling

mhsdesign opened the pull request:
With Neos 9 Beta 8 the generic concept of subtree tags was introduced: https://github.com/neos/neos-development-collection/pull/4659 Disabling publishes since then `SubtreeWasTagged`.

It stated:

> To the core, Subtree tags have no special meaning. With the exception of the disabled tag that leads to the affected nodes to be excluded from subgraph queries in the frontend

This change adjust this to that there are no subtree tags with special meaning in the core.

The duplicated command implementation will be removed with Neos 9 Beta 19 and its now discouraged to use these legacy commands which now translate fully to their subtree counterparts.

The notion of the special disabled tag `SubtreeTag::disabled()` is thus now fully deprecated in the core instead `NeosSubtreeTag::disabled()` must be used.

<!-- 
    Thanks for your contribution, we appreciate it!

    The first section should explain briefly what is changed. 
    Some examples are always nice to showcase the use. 
    The content will be used in the change-logs and addresses 
    developers working with Neos.

    If there are issues regarding the topic of your PR link 
    them here as `related:` or `resolved:`
-->

**Upgrade instructions**

<!-- 
    Add upgrade instructions for breaking changes. 
    Explain who is affected, what has to be adjusted  
-->

**Review instructions**

<!-- 
    If your change is not explained fully by the first section you can 
    add more details here to help the reviewers understand the change.
    We have to understand what you did, why you did it and how we can 
    verify it works correctly and does no harm.
-->

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

mhsdesign committed:
TASK: Discourage `DisableNodeAggregate` and remove duplicated command handling

With Neos 9 Beta 8 the generic concept of subtree tags was introduced: https://github.com/neos/neos-development-collection/pull/4659
Disabling publishes since then `SubtreeWasTagged`.

The duplicated command implementation will be removed with Neos 9 Beta 19 and its now discouraged to use these legacy commands
which now translate fully to their subtree counterparts.

bwaidelich reviewed (approved):
I really like this, +1 by skimming over :)

kitsunet reviewed (approved):
yep great

mhsdesign committed:
Merge remote-tracking branch 'origin/9.0' into task/deprecate-DisableNodeAggregate

mhsdesign committed:
TASK: Introduce `NeosSubtreeTag::disabled`

And deprecate `SubtreeTag::disabled`

see discussion in https://github.com/neos/neos-development-collection/pull/5463#discussion_r1981204343

The only behavioural change is that with that the content repository does not know any subtree tag by default and thus the new default of the `StaticAuthProvider::getVisibilityConstraints` is to have nothing excluded!
